extensions/binding/org.eclipse.smarthome.binding.astro.test/src/test/groovy/org/eclipse/smarthome/binding/astro/test/AstroOSGiTest.groovy

```diff
@@ -0,0 +1,246 @@
+import org.eclipse.smarthome.core.thing.link.ManagedItemChannelLinkProvider
+import org.eclipse.smarthome.core.types.State
+import org.eclipse.smarthome.core.types.UnDefType
+import org.eclipse.smarthome.test.OSGiTest
+import org.eclipse.smarthome.test.storage.VolatileStorageService
+import org.junit.After
+import org.junit.AfterClass
+import org.junit.Before
+import org.junit.BeforeClass
+import org.mockito.Mockito
+
+/**
+ * The base class for the OSGi tests for the Astro Binding.
+ *
+ * @author Petar Valchev - Initial Implementation
+ * @author Svilen Valkanov - Added missing configuration parameters and replaced headers
+ */
+class AstroOSGiTest extends OSGiTest {
+    protected ManagedThingProvider managedThingProvider
+    protected ThingRegistry thingRegistry
+    protected ItemRegistry itemRegistry
+
+    protected static final String TEST_SUN_THING_ID = "testSunThingId"
+    protected static final String TEST_MOON_THING_ID = "testMoonThingId"
+    protected final String TEST_ITEM_NAME = "testItem"
+
+    protected final String DEFAULT_TEST_CHANNEL_ID = "rise#start"
+    protected final String GEOLOCATION_PROPERTY = "geolocation"
+    protected final String GEOLOCATION_VALUE = "51.2,25.4"
+    protected final String INTERVAL_PROPERTY = "interval"
+    protected final BigDecimal INTERVAL_DEFAULT_VALUE = new BigDecimal(300)
+
+    protected AstroThingHandler astroHandler
+    protected Thing astroThing
+    protected GenericItem testItem
+
+    private static int refreshJobsDelay
+
+    public enum AcceptedItemType{
+        DATE_TIME("DateTime"), STRING("String"), NUMBER("Number");
+
+        private String acceptedItemType
+
+        public AcceptedItemType(String acceptedItemType){
+            this.acceptedItemType = acceptedItemType
+        }
+
+        public String getAcceptedItemType(){
+            return acceptedItemType
+        }
+    }
+
+    @BeforeClass
+    public static void setUpClass(){
+        refreshJobsDelay = AstroThingHandler.refreshJobsDelay
+        AstroThingHandler.refreshJobsDelay = 0
+    }
+
+    @Before
+    public void setUp(){
+        VolatileStorageService volatileStorageService = new VolatileStorageService()
+        registerService(volatileStorageService)
+
+        managedThingProvider = getService(ThingProvider, ManagedThingProvider)
+        assertThat "Could not get ManagedThingProvider",
+                managedThingProvider,
+                is(notNullValue())
+
+        thingRegistry = getService(ThingRegistry)
+        assertThat "Could not get ThingRegistry",
+                thingRegistry,
+                is(notNullValue())
+
+        itemRegistry = getService(ItemRegistry)
+        assertThat "Could not get ItemRegistry",
+                itemRegistry,
+                is(notNullValue())
+                
+    }
+
+    @After
+    public void tearDown(){
+        if(astroThing != null){
+            Thing removedThing = thingRegistry.forceRemove(astroThing.getUID())
```

Comment:
Your are correctly using the `managedThingProvider` to add the thing. Why don't you remove it the same way?